zyte_common_items/base.py

```diff
@@ -91,30 +93,76 @@ def _apply_field_types_to_sub_fields(cls, item: Optional[Dict]):
 
         from_dict, from_list = {}, {}
 
-        annotations = getattr(cls, "__annotations__", {})
+        annotations = ChainMap(*(c.__annotations__ for c in cls.__mro__ if "__annotations__" in c.__dict__))
 
         for field, type_annotation in annotations.items():
             origin = get_origin(type_annotation)
-
-            if origin == list:
-                from_list[field] = get_args(type_annotation)[0]
-
-            elif origin == Union:
+            if origin == Union:
                 field_classes = get_args(type_annotation)
                 if len(field_classes) != 2 or not isinstance(None, field_classes[1]):
                     raise ValueError("Field should only be annotated with one type (or optional).")
+                type_annotation = field_classes[0]
+                origin = get_origin(type_annotation)
 
-                field_class = field_classes[0]
-
-                # ignore classes like str, float, int etc
-                if not is_data_container(field_class):
-                    continue
-
-                from_dict[field] = field_class
+            if origin is list:
+                type_annotation = get_args(type_annotation)[0]
+                if is_data_container(type_annotation):
+                    from_list[field] = type_annotation
+            elif is_data_container(type_annotation):
+                from_dict[field] = type_annotation
 
         if from_dict or from_list:
             item = dict(**item)
             item.update({key: cls.from_dict(item.get(key)) for key, cls in (from_dict or {}).items()})
             item.update({key: cls.from_list(item.get(key, [])) for key, cls in (from_list or {}).items()})
 
         return item
+
+
+@export
+@attrs.define(slots=True)
+class AdditionalProperty(Item):
+    name: str
+    value: str
+
+
+@export
+@attrs.define(slots=True, kw_only=True)
+class AggregateRating(Item):
+    bestRating: Optional[float] = None
+    ratingValue: Optional[float] = None
+    reviewCount: Optional[int] = None
+
+
+@export
+@attrs.define(slots=True)
+class Brand(Item):
+    name: str
+
+
+@export
+@attrs.define(slots=True, kw_only=True)
+class Breadcrumb(Item):
+    name: Optional[str] = None
+    link: Optional[str] = None
+
+
+@export
+@attrs.define(slots=True)
+class Gtin(Item):
+    type: str
+    value: str
+
+
+@export
+@attrs.define(slots=True, kw_only=True)
+class Image(Item):
+    dataUrl: Optional[str] = None
```

Comment:
we decided to drop dataUrl fields from everywhere. 
"url" field shouldn't contain data urls (they need to be filtered out); validation is out of scope for this PR, we can handle it separately.